avm/res/devops-infrastructure/managed-devops-pools/main.bicep

```diff
@@ -0,0 +1,448 @@
+@description('Required. Name of the pool. It needs to be globally unique.')
+param name string
+
+@description('Required. The Azure SKU name of the machines in the pool.')
+param fabricProfileSkuName string
+
+@minValue(1)
+@maxValue(10000)
+@description('Required. Defines how many resources can there be created at any given time.')
+param concurrency int
+
+@description('Required. The VM images of the machines in the pool.')
+param images imageType
+
+@description('Optional. The geo-location where the resource lives.')
+param location string = resourceGroup().location
+
+@description('Required. The resource id of the DevCenter Project the pool belongs to.')
+param devCenterProjectResourceId string
+
+@description('Optional. The subnet id on which to put all machines created in the pool.')
+param subnetId string?
+
+@description('Optional. The secret management settings of the machines in the pool.')
+#disable-next-line secure-secrets-in-params
+param secretsManagementSettings secretManagementType
+
+@description('Optional. A list of empty data disks to attach.')
+param dataDisks dataDiskType
+
+@description('Optional. The Azure SKU name of the machines in the pool.')
+@allowed([
+  'Premium'
+  'Standard'
+  'StandardSSD'
+])
+param osDiskStorageAccount string?
+
+@description('Optional. Determines how the service should be run. By default, this will be set to Service.')
+@allowed([
+  'Interactive'
+  'Service'
+])
+param logonType string?
+
+@description('Required. Defines how the machine will be handled once it executed a job.')
+param agentProfile agentProfileType
+
+@description('Required. Defines the organization in which the pool will be used.')
+param organizationProfile organizationProfileType
+
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Optional. The lock settings of the service.')
+param lock lockType
+
+@description('Optional. Array of role assignments to create.')
+param roleAssignments roleAssignmentType
```

Comment:
Please note that the spec was recently updated and so should this module before being merged. Ref: https://azure.github.io/Azure-Verified-Modules/specs/shared/interfaces/#role-assignments

Reply:
Looks clean!